runtime/vm/createramclass.cpp

```diff
@@ -2921,10 +2922,31 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 	J9Module* module = NULL;
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 	UDATA romFieldCount = romClass->romFieldCount;
-	UDATA flattenedClassCacheAllocSize = sizeof(J9FlattenedClassCache) * (romFieldCount + 1);
+	/*             Structure of Flattened Class Cache              */
+	/***************************************************************
+	 *                       |              |                      *  
+	 *      Default Value    |    empty     |        size          *
+	 *_______________________|______________|______________________* 
+	 *                       |              |                      *
+	 *        clazz1         |     NaS1     |       offset1        *
+	 *_______________________|______________|______________________* 
+	 *                       |              |                      * 
+	 *           *           |      *       |           *          *            
+	 *           *           |      *       |           *          * 
+	 *           *           |      *       |           *          * 
+	 *                       |              |                      *
+	 *        clazzN         |     NaSN     |       offsetN        *
+	 ***************************************************************/		
+	UDATA flattenedClassCacheAllocSize = SIZEOF_FLATTENED_CLASS_CACHE_HEADER + sizeof(J9FlattenedClassCacheEntry) * romFieldCount;
 	UDATA largestFieldType = 0;
-	/* The +1 is for the size field */
-	J9FlattenedClassCache flattenedClassCacheBuffer[DEFAULT_SIZE_OF_CLASS_CACHE + 1] = {0};
+	/*
+	 * Since the header and entry are of different size
+	 * they need to be created separately
+	 * the [1] is the number of header row
+	 */
+	J9FlattenedClassCache flattenedClassCacheBuffer[1];
+	J9FlattenedClassCacheEntry flattenedClassCacheEntryBuffer[DEFAULLT_NUMBER_OF_ENTRIES_IN_FLATTENED_CLASS_CACHE] = {0};
+	flattenedClassCacheBuffer->entry = flattenedClassCacheEntryBuffer;
 	J9FlattenedClassCache *flattenedClassCache = flattenedClassCacheBuffer;
```

Comment:
@tajila I am stuck at this point when trying to make the entry array to size of 
DEFAULLT_NUMBER_OF_ENTRIES_IN_FLATTENED_CLASS_CACHE.
since the entry array has already got one element in it, I cannot directly assign the buffer, 
which is of size 8, to the entry array.

Reply:
@tajila the problem is specifically at line 2949

Reply:
change it to

```
UDATA flattenedClassCacheAllocSize = SIZEOF_FLATTENED_CLASS_CACHE_HEADER + (sizeof(J9FlattenedClassCacheEntry) * romFieldCount);
U_8 flattenedClassCacheEntryBuffer[SIZEOF_FLATTENED_CLASS_CACHE_HEADER + (sizeof(J9FlattenedClassCacheEntry) * DEFAULLT_NUMBER_OF_ENTRIES_IN_FLATTENED_CLASS_CACHE];
J9FlattenedClassCache *flattenedClassCache = (J9FlattenedClassCache *) flattenedClassCacheBuffer;